src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1311,6 +1311,29 @@ private static Type GetCommonType(Type minType, Type maxType)
 
             return resultType;
         }
+
+        /// <summary>
+        /// Returns a list containing only the elements that passed the attribute's validation.
+        /// </summary>
+        /// <param name="elementsToValidate">The objects to validate.</param>
+        internal IList<T> GetValidatedElements<T>(IList<T> elementsToValidate)
+        {
+            var result = new List<T>(elementsToValidate.Count);
+            foreach (var el in elementsToValidate)
+            {
+                try
+                {
+                    ValidateElement(el);
+                    result.Add(el);
+                }
+                catch (ValidationMetadataException)
+                {
+                    // Element was not in range - drop
+                }
+            }
+
+            return result;
+        }
```

Comment:
```suggestion
        internal static IEnumerable GetValidatedElements(IEnumerable elementsToValidate)
        {
            foreach (var element in elementsToValidate)
            {
                try
                {
                    ValidateElement(element );
                    yield return element;
                }
                catch (ValidationMetadataException)
                {
                    // Element was not valid - drop
                }
            }
        }
```

Reply:
`IList` was used to be explicit about order being kept (and initially `for`-loop), but will change. 👍

Assuming `static` was typo. 🙂